programs/nft-voter/src/instructions/cast_nft_vote.rs

```diff
@@ -0,0 +1,105 @@
+use anchor_lang::prelude::*;
+use anchor_lang::{Accounts};
+use itertools::Itertools;
+use spl_governance_addin_api::voter_weight::VoterWeightAction;
+use spl_governance_tools::account::create_and_serialize_account_signed;
+use crate::{state::*, id};
+use crate::error::NftVoterError;
+
+/// Casts NFT vote. The NFTs used for voting are tracked using NftVoteRecord accounts
+/// This instruction updates VoterWeightRecord which is valid for the current Slot and the target Proposal only
+/// and hance the instruction has to be executed inside the same transaction as spl-gov CastVote
+#[derive(Accounts)]
+#[instruction(proposal: Pubkey)]
+pub struct CastNftVote<'info> {
```

Comment:
nitpick: castvote name is slightly misleading imo, afaics there is no type of vote in this ix - yay/nay, so user is not casting his voter per say, rather just accumulating his voting power, no?

Reply:
This is tricky. It's still a vote but not directional. The instruction doesn't record vote choice (ex. Yes/No) but it's nonetheless a vote for each individual NFT. 

I also wanted to keep it symmetrical with `spl-gov`:  `cast_nft_vote` is always called together with `spl-gov.cast_vote` in the same tx and `relinquish_nft_vote` with `spl-gov.relinquish_vote`

I updated comments with more explanations how the instruction is used